src/Toml/Bi/Combinators.hs

```diff
@@ -217,12 +213,16 @@ arrayIntSet = match _IntSet
 
 -- | Parser for hash sets. Takes converter for single hashable value and
 -- returns a set of hashable values.
-arrayHashSetOf :: (Typeable a, Hashable a, Eq a) => BiMap a AnyValue -> Key -> TomlCodec (HashSet a)
+arrayHashSetOf
+    :: (Typeable a, Hashable a, Eq a)
+    => TomlBiMap a AnyValue
+    -> Key
+    -> TomlCodec (HashSet a)
 arrayHashSetOf = match . _HashSet
 
 -- | Parser for non- empty lists of values. Takes converter for single value and
 -- returns a non-empty list of values.
-arrayNonEmptyOf :: Typeable a => BiMap a AnyValue -> Key -> TomlCodec (NonEmpty a)
+arrayNonEmptyOf :: (a ~ Value t, Typeable a) => TomlBiMap a AnyValue -> Key -> TomlCodec (NonEmpty a)
```

Comment:
Otherwise an error returns:
```
Couldn't match type ‘a’ with ‘Value t0’
    arising from a use of ‘_NonEmpty’
...
```

Reply:
Hmm, that's strange :thinking: I will try to investigate this.

Reply:
@willbasky Okay, I've checked. This is because of type of `_NonEmpty`:

```haskell
_NonEmpty :: a ~ Value t => TomlBiMap a AnyValue -> TomlBiMap (NE.NonEmpty a) AnyValue
```

The `a ~ Value t` constraint is wrong here. The implementation of `_NonEmpty` should be look kike `_Array`